scalyr_agent/monitor_utils/k8s.py

```diff
@@ -566,42 +580,209 @@ def process_object( self, obj ):
+        @return: True if new filters need to be created based on the differences between the current and the new config.
+                 False otherwise
+        """
+        return self.namespaces_to_ignore != new_config.namespaces_to_ignore
+
+
 class KubernetesCache( object ):
 
-    def __init__( self, k8s, logger, cache_expiry_secs=30, cache_purge_secs=300, namespaces_to_ignore=None ):
+    def __init__( self, api_url="https://kubernetes.default", verify_api_queries=True, cache_expiry_secs=30, cache_purge_secs=300, namespaces_to_ignore=None, start_caching=True ):
+
+        self._lock = threading.Lock()
+        self._namespace_filter = None
+        self._node_filter = None
+        self._k8s = None
+        self._cache_expiry_secs=cache_expiry_secs
+        self._cache_purge_secs=cache_purge_secs
+        self._config_update_count = 0
+
+        empty_config = _CacheConfig( api_url='', namespaces_to_ignore=[] )
 
-        self._k8s = k8s
 
-        self._namespace_filter = self._build_namespace_filter( namespaces_to_ignore )
-        self._node_filter = self._build_node_filter( self._namespace_filter )
+        new_config = _CacheConfig( api_url=api_url,
+                                   verify_api_queries=verify_api_queries,
+                                   cache_expiry_secs=cache_expiry_secs,
+                                   cache_purge_secs=cache_purge_secs,
+                                   namespaces_to_ignore=namespaces_to_ignore )
+
+        # perform initial configuration of the cache
+        self._configure( old_config=empty_config, new_config=new_config )
 
         # create the controller cache
-        self._controller_processor = ControllerProcessor( logger )
-        self._controllers = _K8sCache( logger, k8s, self._controller_processor, '<controller>',
-                               filter=self._namespace_filter,
+        self._controller_processor = ControllerProcessor()
+        self._controllers = _K8sCache( self._controller_processor, '<controller>',
                                perform_full_updates=False )
 
         # create the pod cache
-        self._pod_processor = PodProcessor( logger, self._controllers )
-        self._pods = _K8sCache( logger, k8s, self._pod_processor, 'Pod',
-                               filter=self._node_filter )
+        self._pod_processor = PodProcessor( self._controllers )
+        self._pods = _K8sCache( self._pod_processor, 'Pod' )
 
         self._cluster_name = None
-        self._cache_expiry_secs = cache_expiry_secs
-        self._cache_purge_secs = cache_purge_secs
         self._last_full_update = time.time() - cache_expiry_secs - 1
 
-        self._lock = threading.Lock()
         self._initialized = False
 
-        self._thread = StoppableThread( target=self.update_cache, name="K8S Cache" )
+        self._thread = None
+
+        if start_caching:
+            self.start()
+
+    def _configure( self, old_config, new_config ):
```

Comment:
I don't think you need to be passing in `old_config` here.  Just use what is in `self._config` as the current config.

Reply:
See previous comments.